### PR TITLE
fix: change the /version to /apisix/admin/tool/version

### PR DESCRIPTION
**Pull request by starsz**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
- 
https://github.com/apache/apisix-dashboard/pull/1408#discussion_r570317906

### New feature or improvement
- Describe the details and related test reports.

change the /version to /apisix/admin/tool/version

### CODE REVIEW

**Review by nic-chen — APPROVED** [review]

File: api/internal/filter/authentication.go [hunk 1]
```diff
@@ -45,50 +45,55 @@ func (mw *AuthenticationMiddleware) Handle(ctx droplet.Context) error {
...
 
+	// Need check the auth header
+	tokenStr := req.Header.Get("Authorization")
+
+	// verify token
+	token, err := jwt.ParseWithClaims(tokenStr, &jwt.StandardClaims{}, func(token *jwt.Token) (interface{}, error) {
+		return []byte(conf.AuthConf.Secret), nil
+	})
+
+	// TODO: design the response error code
```
Comment (juzhiyuan) [inline api/internal/filter/authentication.go:64]: may have a tracker issue

**Comment by juzhiyuan** [comment]
ping @tokers @imjoey 's help to review codes.

File: api/internal/filter/authentication.go [hunk 2]
```diff
@@ -45,50 +45,55 @@ func (mw *AuthenticationMiddleware) Handle(ctx droplet.Context) error {
...
-		if _, ok := conf.UserList[claims.Subject]; !ok {
-			log.Warnf("user not exists by token claims subject %s", claims.Subject)
-			ctx.SetOutput(&data.SpecCodeResponse{StatusCode: http.StatusUnauthorized, Response: response})
-			return nil
-		}
+	if req.URL.Path == "/apisix/admin/tool/version" || req.URL.Path == "/apisix/admin/user/login" {
+		return mw.BaseMiddleware.Handle(ctx)
+	}
 
+	if !strings.HasPrefix(req.URL.Path, "/apisix") {
```
Comment (tokers) [inline api/internal/filter/authentication.go:52]: What about swapping these two if block? URI without /apisix can be judged firstly.

**Review by tokers — APPROVED** [review]

File: api/internal/filter/authentication.go (hunk 1 above)
Comment (starsz) [reply api/internal/filter/authentication.go:64]: Yes.It's created already.

File: api/internal/filter/authentication.go (hunk 2 above)
Comment (starsz) [reply api/internal/filter/authentication.go:52]: Good capture. I will do this in another PR.

File: api/internal/filter/authentication.go (hunk 1 above)
Comment (juzhiyuan) [reply api/internal/filter/authentication.go:64]: Could you reference this PR with that issue? to prevent from missing this case.

Comment (starsz) [reply api/internal/filter/authentication.go:64]: See the PR https://github.com/apache/apisix-dashboard/issues/758

Comment (juzhiyuan) [reply api/internal/filter/authentication.go:64]: image

OK, once this issue occurred here once, then that issue will have the ability to track this PR.

**Pull request merged**